main/CMakeLists.txt

```diff
@@ -28,11 +28,16 @@ endif()
 if(proof)
   ROOT_EXECUTABLE(proofserv.exe pmain.cxx LIBRARIES Core MathCore)
 endif()
+
+
 if(MSVC)
   ROOT_EXECUTABLE(hadd hadd.cxx LIBRARIES Core RIO Net Hist Graf Graf3d Gpad Tree Matrix MathCore)
   set_property(TARGET hadd APPEND_STRING PROPERTY LINK_FLAGS "setargv.obj")
 else()
   ROOT_EXECUTABLE(hadd hadd.cxx LIBRARIES Core RIO Net Hist Graf Graf3d Gpad Tree Matrix MathCore MultiProc)
+  if (CMAKE_COMPILER_IS_GNUCC AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.0)
+    target_link_libraries(hadd PUBLIC stdc++fs)
```

Comment:
```suggestion
    target_link_libraries(hadd stdc++fs)
```
otherwise we get:
```
2024-11-12T08:38:35.2925168Z CMake Error at main/CMakeLists.txt:39 (target_link_libraries):
2024-11-12T08:38:35.2926143Z   The plain signature for target_link_libraries has already been used with
2024-11-12T08:38:35.2927218Z   the target "hadd".  All uses of target_link_libraries with a target must be
2024-11-12T08:38:35.2928320Z   either all-keyword or all-plain.
2024-11-12T08:38:35.2928666Z 
2024-11-12T08:38:35.2928861Z   The uses of the plain signature are here:
2024-11-12T08:38:35.2929243Z 
2024-11-12T08:38:35.2929739Z    * cmake/modules/RootMacros.cmake:1408 (target_link_libraries)
```